dmoj/executors/RUST.py

```diff
@@ -56,15 +59,44 @@ def create_files(self, problem_id, source_code, *args, **kwargs):
         with open(self._file('Cargo.toml'), 'wb') as f:
             f.write(CARGO_TOML)
 
+    shared_target = None
+
+    @classmethod
+    def get_shared_target(cls):
+        if cls.shared_target is None:
+            cargo_dir = os.path.expanduser('~/.cargo')
+            collisions = 0
+            judge_id = env.get('id', 'anonymous')
+            hasher = hashlib.sha256()
+            hasher.update(judge_id.encode('utf-8'))
+            id_hash = hasher.hexdigest()
+            while True:
+                maybe_target = cargo_dir + f'/dmoj-shared-target-{id_hash}-{collisions}'
+                try:
+                    os.mkdir(maybe_target, mode=0o775)
+                except FileExistsError:
+                    pass
+                try:
+                    # This is 'leaked' on purpose.
+                    dirfd = os.open(maybe_target, os.O_RDONLY | os.O_CLOEXEC)
+                    fcntl.flock(dirfd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+                except BlockingIOError:
```

Comment:
Missing increment of collisions (also failing mypy elsewhere)

Reply:
Can you merge #1078 to fix mypy?

Reply:
Also thanks, fixed.